### PR TITLE
Implement save location

### LINKED ISSUE #84: Change save path

### PR DESCRIPTION
**Pull request by turtle96**
This closes #84

### CODE REVIEW

**Comment by turtle96** [comment]
Ready for review
@CS2103AUG2016-W15-C3/code-quality 
@CS2103AUG2016-W15-C3/documentation 
@CS2103AUG2016-W15-C3/testing

File: docs/UserGuide.md [hunk 1]
```diff
@@ -119,11 +119,13 @@ Format: `edit INDEX NEWTASK`<br>
...
 
-#### Saving the information in Taskell
+#### Saving the information in Taskell : `save`
 
-Specify the path of a new folder to store the data file of Taskell. The user should have permissions to access the folder.<br>
-Format: `save /FOLDERPATH`<br>
-Example: `save /Jim/Documents`
+Specify the path of a new folder to store the data file of Taskell. The user should have permissions to access the folder. <br>
+If directory is valid but file is missing, for example if command is `save C:/Users/Jim/Documents/chicken`,  and `C:/Users/Jim/Documents/chicken` is valid
+but `chicken` file is not created, Taskell will create the file.<br>
```
Comment (marcusngwenjian) [inline docs/UserGuide.md]: Suggestion (Use less of passive words):
<br>
Specify the path <strong>to a new folder where you want</strong> ~~of a new folder~~ to store ~~the~~ <strong>your</strong> data file of Taskell. ~~The user~~ <strong>You</strong> should have permission~~s~~to access ~~the~~ <strong>this</strong> folder. <br>
 Format: `save /FOLDERPATH`<br>
If <strong>the specified</strong> directory is valid but <strong>the</strong> file is missing, for example if command is `save C:/Users/Jim/Documents/chicken`,  and `C:/Users/Jim/Documents/chicken` is valid but `chicken` file is not created, Taskell will create the file <strong>for you</strong>.<br>
 Example: `save /Jim/Documents`

File: src/main/java/seedu/taskell/commons/core/Config.java [hunk 2]
```diff
@@ -61,7 +61,6 @@ public void setTaskManagerName(String taskManagerName) {
         this.taskManagerName = taskManagerName;
     }
 
-
     @Override
     public boolean equals(Object other) {
         if (other == this){
```
Comment (marcusngwenjian) [inline src/main/java/seedu/taskell/commons/core/Config.java]: Space is needed after brackets <br>

image

File: src/main/java/seedu/taskell/storage/JsonConfigStorage.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+ * A class to access Config stored in the hard disk as a json file
+ */
+
+public class JsonConfigStorage implements ConfigStorage {
+
+    private static final Logger logger = LogsCenter.getLogger(JsonConfigStorage.class);
+    
+    private String filePath;
+    
+    public JsonConfigStorage(String filePath){
```
Comment (marcusngwenjian) [inline src/main/java/seedu/taskell/storage/JsonConfigStorage.java]: Space is needed
image

File: src/test/java/guitests/SaveStorageLocationCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        JsonConfigStorage jsonConfigStorage = new JsonConfigStorage(CONFIG_LOCATION);
+        Optional<Config> config = jsonConfigStorage.readConfig(CONFIG_JSON);
+        assert(config.isPresent());
+    } 
+    
+    private void assertWriteToXmlSuccess() {
+        TestTask[] currentList = td.getTypicalTasks();
+        assertTrue(taskListPanel.isListMatching(currentList));
+    }
+
```
Comment (marcusngwenjian) [inline src/test/java/guitests/SaveStorageLocationCommandTest.java]: Extra newline can be removed

**Comment by marcusngwenjian** [comment]
Code Quality reviewed

**Comment by marcusngwenjian** [comment]
Code Quality reviewed

**Comment by turtle96** [comment]
made edits for code quality

**Comment by Mamtha3005** [comment]
Documentation Reviewed

**Comment by turtle96** [comment]
@ZuoZhuolin please test

**Comment by ZuoZhuolin** [comment]
Test reviewed

**Pull request merged**